fix: assert group(partition) key type during physical plan compilation

aceforeverd opened the pull request:
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)


* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

fix: #613 

in `performance sensitive mode` = false, group key type is not checked. And, allow unsupport key like float as partition key.

Basicly, the runner only support partition key is
- bool
- intxxx
- string
- date
- timestamp

any other type will result compilation error

aceforeverd committed:
fix: #613

aceforeverd commented on hybridse/src/passes/physical/group_and_sort_optimized.cc:
```diff
@@ -454,6 +459,27 @@ bool GroupAndSortOptimized::TransformKeysAndOrderExpr(
+
+                size_t schema_idx = 0;
+                size_t col_idx = 0;
+                if (root_schemas_ctx->ResolveColumnRefIndex(column, &schema_idx, &col_idx).isOK()) {
```
window & last join & group

aceforeverd replied:
Status BatchModeTransformer::CheckWindow
do not add in group_and_sort_optimized

aceforeverd committed:
Merge remote-tracking branch 'upstream/main' into fix/613-assert-partition-key-type

aceforeverd committed:
fix: #613

add `CheckPartitionColumn` in transformer

aceforeverd committed:
test: add ut for #613

aceforeverd committed:
fix: #613

add key check for GROUP node

jingchen2222 commented on hybridse/include/sdk/base.h:
```diff
@@ -72,6 +73,34 @@ inline const std::string DataTypeName(const DataType& type) {
+    }
+    return "unknown";
+}
+
```
We won't expose fe_type.ph.h to SDK users. That's why we need `sdk::DataType`. 
Move `inline const std::string TypeName(type::Type type) ` to somewhere else like include/node/type_node.h or include/node/sql_node.h

BTW, we are going to refactor prototypes in hybridse and openmldb, which means this function might be rewritten in the future.  So we don't have to worry too much about if it's appropriate to place `TypeName (type::Type type)` in include/node/xxx.h.

aceforeverd replied:
cool. BTW there is `sdk::Type_Name` provided by protobuf, but I'd like to rm the `k` prefix

jingchen2222 commented on hybridse/src/vm/transform.cc:
```diff
@@ -1751,6 +1753,45 @@ Status BatchModeTransformer::CheckTimeOrIntegerOrderColumn(const node::OrderByNo
+                        case type::kTimestamp:
+                            break;
+                        default: {
+                            CHECK_TRUE(false, common::kPhysicalPlanError, "unsupported group key, type is ",
```
Use FAIL_STATUS(common::kPhysicalPlanError, msg) instead.

jingchen2222 commented on hybridse/src/passes/physical/group_and_sort_optimized.cc:
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 #include "passes/physical/group_and_sort_optimized.h"
+#include <absl/strings/str_cat.h>
```
unused, remove?

aceforeverd committed:
test: fix two ut in `transform_request_mode_test`

aceforeverd committed:
fix: #613

- assert last join op's right keys for supported type
- fix few review comments
- a few code format

aceforeverd committed:
Merge remote-tracking branch 'upstream/main' into fix/613-assert-partition-key-type

aceforeverd committed:
fix: #613

- add partition type check for last join node

aceforeverd committed:
test: add test for parition type

aceforeverd committed:
chore: improve error message in runer

aceforeverd committed:
test: update `transform_test.sh`

aceforeverd committed:
Merge remote-tracking branch 'upstream/main' into fix/613-assert-partition-key-type

aceforeverd committed:
doc: update comment

aceforeverd committed:
Merge remote-tracking branch 'upstream/main' into fix/613-assert-partition-key-type

aceforeverd committed:
fix: #613

add check for `join_list` in project node

aceforeverd committed:
Merge remote-tracking branch 'upstream/main' into fix/613-assert-partition-key-type

jingchen2222 reviewed (approved):
LGTM

aceforeverd committed:
test: fix ddl_parser_test test

avoid write sql case whose partition key is float/double type

jingchen2222 reviewed (approved):
LGTM

vagetablechicken reviewed (approved):
LGTM
